src/hooks/WebRTC/useWebRTC.ts

```diff
@@ -16,50 +16,33 @@ export function useWebRTC(signalUrl: string, configuration?: RTCConfiguration) {
         peer.current.ontrack = handleTrack;
 
         signalChannel.current = new SignalChannel(signalUrl);
-        signalChannel.current.addSignalListener("answer", handleAnswerSignal);
-        signalChannel.current.addSignalListener(
-            "candidate",
-            handleCandidateSignal
-        );
-
-        signalChannel.current.socket.onopen = () => startHandshake();
+        signalChannel.current.addSignalListener("offer", handleOfferSignal);
+        signalChannel.current.addSignalListener("candidate", handleCandidateSignal);
     }, []);
 
-    function startHandshake() {
-        console.log("start handshake");
-
-        peer.current!.createOffer().then(
-            (offer: RTCSessionDescriptionInit) => sendAndUpdateOffer(offer),
-            (reason: Error) =>
-                signalChannel.current!.sendClose(
-                    SignalCode.FailCreateOffer,
-                    reason.message
-                )
-        );
-    }
-
-    function sendAndUpdateOffer(offer: RTCSessionDescriptionInit) {
-        peer.current!.setLocalDescription(offer).then(
-            () => signalChannel.current!.sendSignal("offer", offer),
-            (reason: Error) =>
-                signalChannel.current!.sendClose(
-                    SignalCode.FailUpdateLocalOffer,
-                    reason.message
+    function sendAndUpdateAnswer() {
+        peer.current!.createAnswer().then(
+            (answer) => {
+                peer.current!.setLocalDescription(answer).then(
```

Comment:
instead of nested `.then()`, return `peer.currrent!.setLocalDescription(answer)` and then catch it in the next then